projectInit/init.go

```diff
@@ -99,169 +101,110 @@ func initializeTemplate(templateName string) error {
+// FromURL initializes a Gauge project with specified template URL
+func fromURL(templateURL string, silent bool) {
+	checkURL(templateURL)
+	if err := initializeTemplate(templateURL); err != nil {
+		logger.Fatalf(true, "Failed to initialize project. %s", err.Error())
 	}
-	return nil
+	installRunner(silent)
 }
 
-func createOrAppendGitignoreFile() error {
-	destFile := filepath.Join(gitignoreFileName)
-	srcFile, err := common.GetSkeletonFilePath(gitignoreFileName)
+// fromZipFile initializes a Gauge project with specified zip file
+func fromZipFile(templateFile string, silent bool) {
+	validateDirectory()
+	tempDir := common.GetTempDir()
+	defer util.Remove(tempDir)
+	unzippedTemplateDir, err := common.UnzipArchive(templateFile, tempDir)
 	if err != nil {
-		showMessage("error", fmt.Sprintf("Failed to read .gitignore file. %s", err.Error()))
-		return err
+		logger.Fatalf(true, "Failed to initialize project. %s", err.Error())
 	}
-	showMessage("create", destFile)
-	if err := common.AppendToFile(srcFile, destFile); err != nil {
-		showMessage("error", err.Error())
-		return err
+	err = copyTemplateContents(unzippedTemplateDir)
+	if err != nil {
+		logger.Fatalf(true, "Failed to initialize project. %s", err.Error())
 	}
-	return nil
+	installRunner(silent)
 }
 
-func createSkeletonFile() error {
-	skelFile, err := common.GetSkeletonFilePath(skelFileName)
+func validateDirectory() {
+	wd, err := os.Getwd()
 	if err != nil {
-		return err
+		logger.Fatalf(true, "Failed to find working directory. %s", err.Error())
 	}
-	specFile := filepath.Join(specsDirName, skelFileName)
-	showMessage("create", specFile)
-	if common.FileExists(specFile) {
-		showMessage("skip", specFile)
-		return err
-	} else {
-		err = common.CopyFile(skelFile, specFile)
-		if err != nil {
-			showMessage("error", fmt.Sprintf("Failed to create %s. %s", specFile, err.Error()))
-			return err
-		}
+	config.ProjectRoot = wd
+	if isGaugeProject() {
+		logger.Fatalf(true, "This is already a Gauge Project. Please try to initialize a Gauge project in a different location.")
 	}
-	return nil
 }
 
-func createSpecDirectory() error {
-	showMessage("create", specsDirName)
-	if !common.DirExists(specsDirName) {
-		err := os.Mkdir(specsDirName, common.NewDirectoryPermissions)
-		if err != nil {
-			showMessage("error", fmt.Sprintf("Failed to create %s. %s", specsDirName, err.Error()))
-			return err
-		}
-	} else {
-		showMessage("skip", specsDirName)
-	}
-	return nil
+func isURL(name string) bool {
+	u, err := url.ParseRequestURI(name)
+	return err == nil && u.Scheme != ""
 }
 
-func createManifestFile(language string) error {
-	showMessage("create", common.ManifestFile)
-	if common.FileExists(common.ManifestFile) {
-		showMessage("skip", common.ManifestFile)
+func checkURL(templateURL string) {
+	u, err := url.ParseRequestURI(templateURL)
+	if err != nil {
+		logger.Fatalf(true, "Failed to parse template URL '%s'. The template location must be a valid (https) URI", templateURL)
```

Comment:
```suggestion
		logger.Fatalf(true, "Failed to parse template URL '%s'. The template location must be a secure and valid  URI", templateURL)
```

As file uri's are allowed too.